src/connect.tsx

```diff
@@ -254,33 +264,31 @@ export default function connect(opts?: ConnectOptions) {
         const { watchQuery, reduxRootKey } = this.client;
         const { store } = this;
 
-        const queryHandles = mapQueriesToProps({
+        const queryOptions = mapQueriesToProps({
           state: store.getState(),
           ownProps: props,
         });
 
         const oldQueries = assign({}, this.previousQueries);
-        this.previousQueries = assign({}, queryHandles);
+        this.previousQueries = assign({}, queryOptions);
 
         // don't re run queries if nothing has changed
-        if (isEqual(oldQueries, queryHandles)) {
+        if (isEqual(oldQueries, queryOptions)) {
           return false;
         } else if (oldQueries) {
           // unsubscribe from previous queries
           this.unsubcribeAllQueries();
         }
 
-        if (isObject(queryHandles) && Object.keys(queryHandles).length) {
-          this.queryHandles = queryHandles;
```

Comment:
As far as I know, this line was not correct - it was assigning query options to the same object we use to hold the observables/handles/subscriptions.